esbuild/esbuild.config.js

```diff
@@ -0,0 +1,87 @@
+const { spawn } = require('node:child_process')
+const path = require('node:path')
+
+const { glob } = require('glob')
+const chokidar = require('chokidar')
+const buildAssets = require('./assets.config')
+const buildApp = require('./app.config')
+
+const cwd = process.cwd()
+
+/**
+ * Configuration for build steps
+ * @type {BuildConfig}
+ */
+const buildConfig = {
+  isProduction: process.env.NODE_ENV === 'production',
+
+  app: {
+    outDir: path.join(cwd, 'dist'),
+    entryPoints: glob
+      .sync([path.join(cwd, '*.ts'), path.join(cwd, 'server/**/*.ts')])
+      .filter(file => !file.endsWith('.test.ts')),
+    copy: [
+      {
+        from: path.join(cwd, 'server/views/**/*'),
+        to: path.join(cwd, 'dist/server/views'),
+      },
+    ],
+  },
+
+  assets: {
+    outDir: path.join(cwd, 'dist/assets'),
+    entryPoints: glob.sync([path.join(cwd, 'assets/js/index.js'), path.join(cwd, 'assets/scss/application.scss')]),
+    copy: [
+      {
+        from: path.join(cwd, 'assets/images/**/*'),
+        to: path.join(cwd, 'dist/assets/images'),
+      },
+      {
+        from: path.join(cwd, 'assets/js/**/*'),
```

Comment:
I've hacked the config for now by copying over the javascript, will add to the `index.js` once this PR merged instead